### PR TITLE
#1069: Added test to Payroll.

### PR DESCRIPTION
**Pull request by paulodamaso**
For #1069:
- Created `PayrollTest` to test `Payroll`
- Created test for amount too small payment;
- Added puzzles to test remaining `Payroll.pay` results;
- Updated puzzle.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1421](https://github.com/zerocracy/farm/issues/1421) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1421](https://github.com/zerocracy/farm/issues/1421) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1940532260); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/zerocracy/pmo/banks/PayrollTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import com.zerocracy.farm.fake.FkFarm;
+import com.zerocracy.farm.fake.FkProject;
+import com.zerocracy.pm.cost.Ledger;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * Tests for {@link Payroll}.
+ * @since 0.25.7
```
Comment (krzyk) [inline src/test/java/com/zerocracy/pmo/banks/PayrollTest.java]: @paulodamaso `0.26`

File: src/main/java/com/zerocracy/pmo/banks/FkBank.java [hunk 2]
```diff
@@ -54,9 +54,9 @@
  * </pre>
  *
  * @since 0.22
- * @todo #565:30min Add FkBank to the Payroll under the file payment method.
- *  Ensure, that the opened files are closed properly and cover Payroll with
- *  tests.
+ * @todo #1069:30min Add FkBank to the Payroll under the file payment method.
+ *  Ensure, that the opened files are closed properly and implement commented
```
Comment (krzyk) [inline src/main/java/com/zerocracy/pmo/banks/FkBank.java]: @paulodamaso Either the tests need to be implemented in this todo or in the ones in tests, not both places.

File: src/test/java/com/zerocracy/pmo/banks/PayrollTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            new Ledger(
+                new FkProject()
+            ),
+            "paulodamaso",
+            amount,
+            reason
+        );
+    }
+
+    // #1069:30min Implement test for wallet is empty check in Payroll. This
```
Comment (krzyk) [inline src/test/java/com/zerocracy/pmo/banks/PayrollTest.java]: @paulodamaso I don't think those 3 todos are big enough to be each in separate todo, I would join 3 of them into a single todo

**Comment by paulodamaso** [comment]
@krzyk Thanks for the review, removed all `PayrollTest` puzzles, please take a look.

**Comment by krzyk** [comment]
@paulodamaso thanks

**Comment by krzyk** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by ypshenychka** [comment]
@krzyk According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

[One](https://github.com/zerocracy/farm/pull/1421#discussion_r202359638) out of three found issues is cosmetic.
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by krzyk** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@krzyk thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**